slf4j-api/src/main/java/org/slf4j/LoggerFactory.java

```diff
@@ -198,8 +198,8 @@ private final static void versionSanityCheck() {
             String requested = StaticLoggerBinder.REQUESTED_API_VERSION;
 
             boolean match = false;
-            for (int i = 0; i < API_COMPATIBILITY_LIST.length; i++) {
-                if (requested.startsWith(API_COMPATIBILITY_LIST[i])) {
+            for (String aAPI_COMPATIBILITY_LIST : API_COMPATIBILITY_LIST) {
```

Comment:
for-each added in Java 5: http://docs.oracle.com/javase/1.5.0/docs/guide/language/foreach.html